### ISSUE TITLE
External optimizer with apply_gradients

### ISSUE THREAD

**Issue by zhengsx** [tensorflow/tensorflow#6927]
Hi all,

I'm trying to define some new optimization algorithm. I found an interface about external optimizer [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/opt/python/training/external_optimizer.py), it says that i need to subclass it and implement _minimize. However I need to call apply_gradients when using the external optimizer. Does anyone know how can i implement it?

Thanks.

**Comment by yaroslavvb** [comment]
Hi, could you please ask this on stackoverflow? We are trying to keep this list for bugs/feature requests, thanks :)

**Comment by zhengsx** [comment]
Sure, thank you for reminding. I put it on [here](http://stackoverflow.com/questions/41724406/how-to-implement-an-external-optimizer-in-tensorflow-with-apply-gradients-method). Hope to get answers :)
